### PR TITLE
Reduce k8s deployment replica and add metrics endpoints

### PR DESCRIPTION
**Pull request by MonkeyCanCode**
# Description

With the original README.md, it indicated the k8s manifest will spin up 2 pods and we will have k8s svc on top. However, this is bad as the default backend metastore is in-memory. Thus, each pod will have their own credential thus the svc became unpredictable (unless we want to do headless one then have user point to each pod implicitly). Thus I reduced the replica to 1 and updated the doc to reflect the same. Along with that, I exposed one more port (8182) that is supported by the catalog to expose jmx metrics as well as healthcheck endpoint. 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

```sh
# Port forward for 8182 for metrics endpoints
kubectl port-forward svc/polaris-service -n polaris 8182:8182

# Show healthcheck endpoint
$ curl -s localhost:8182/healthcheck | jq
{
  "deadlocks": {
    "healthy": true,
    "duration": 0,
    "timestamp": "2024-08-08T23:16:11.064Z"
  },
  "polaris": {
    "healthy": true,
    "duration": 0,
    "timestamp": "2024-08-08T23:16:11.063Z"
  }
}

# Show JMX endpoint
$ curl -s localhost:8182/metrics | head
# HELP polaris_CatalogApi_commitTransaction_count_total
# TYPE polaris_CatalogApi_commitTransaction_count_total counter
polaris_CatalogApi_commitTransaction_count_total 0.0
# HELP polaris_CatalogApi_commitTransaction_error_total
# TYPE polaris_CatalogApi_commitTransaction_error_total counter
polaris_CatalogApi_commitTransaction_error_total{HTTP_RESPONSE_CODE="400"} 0.0
polaris_CatalogApi_commitTransaction_error_total{HTTP_RESPONSE_CODE="500"} 0.0
# HELP polaris_CatalogApi_commitTransaction_seconds
# TYPE polaris_CatalogApi_commitTransaction_seconds summary
polaris_CatalogApi_commitTransaction_seconds_count 0
```

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue
- [x] I have signed and submitted the [ICLA](https://github.com/polaris-catalog/polaris/blob/main/ICLA.md) and if needed, the [CCLA](https://github.com/polaris-catalog/polaris/blob/main/CCLA.md). See [Contributing](https://github.com/polaris-catalog/polaris/blob/main/CONTRIBUTING.md) for details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,8 +56,12 @@ Running in Docker
 - `docker compose up --build --exit-code-from regtest` - To run regression tests in a Docker environment.
 
 Running in Kubernetes
-- `./setup.sh` - To run Polaris as a mini-deployment locally. This will create two pods that bind themselves to port `8181`.
-- `kubectl port-forward svc/polaris-service -n polaris 8181:8181` - To create a secure connection between a local machine and a pod within the cluster.
+- `./setup.sh` - To run Polaris as a mini-deployment locally. This will create one pod that bind itself to ports `8181` and `8182`.
+- `kubectl port-forward svc/polaris-service -n polaris 8181:8181` - To create a secure connection between a local machine and a pod within the cluster for service endpoints.
+- `kubectl port-forward svc/polaris-service -n polaris 8182:8182` - To create a secure connection between a local machine and a pod within the cluster for metrics endpoints.
```
Comment (flyrain) [inline README.md:61]: Can we combine these two commands to one like this?
```
kubectl port-forward svc/polaris-service -n polaris 8181:8181 8182:8182
```

File: k8/deployment.yaml [hunk 2]
```diff
@@ -9,7 +9,6 @@ metadata:
   name: polaris-deployment
   namespace: polaris
 spec:
-  replicas: 2
```
Comment (flyrain) [inline k8/deployment.yaml:12]: Minor suggestion: can we put `1` explicitly in case people don't know the default value?

**Review by flyrain — COMMENTED** [review]
Thanks for the fix. 

Not related to this PR, but I think it would be nice to have a real backend RDMS, i.e. a postgres container.

**Comment by MonkeyCanCode** [comment]
> replicas

Yes, I do like the idea of using a postgres container as the default backend as compared to in-memory (in-memory is good for quick testing for sure but functionality is really limited). However to do so, we will need to decided on couple things:
1. which JDBC driver jar to include as part of the image?
2. Should we have persist backend enable by default (or use a diff config and mount it into the container)?

Here is a similar discussion for this topics: https://github.com/polaris-catalog/polaris/discussions/125

File: README.md (hunk 1 above)
Comment (MonkeyCanCode) [reply README.md]: Took care.

File: k8/deployment.yaml (hunk 2 above)
Comment (MonkeyCanCode) [reply k8/deployment.yaml:12]: Took care.

**Comment by MonkeyCanCode** [comment]
> Thanks for the fix. LGTM overall. Left minor comments:
> 
> Not related to this PR, but I think it would be nice to have a real backend RDMS, i.e. a postgres container.

Yes, I do like the idea of using a postgres container as the default backend as compared to in-memory (in-memory is good for quick testing for sure but functionality is really limited). However to do so, we will need to decided on couple things:

which JDBC driver jar to include as part of the image?
Should we have persist backend enable by default (or use a diff config and mount it into the container)?
Here is a similar discussion for this topics: https://github.com/polaris-catalog/polaris/discussions/125

**Comment by flyrain** [comment]
1. Replied in #125 
2. I think the name `metrics` is fine now. We can always change it in the future if there could be any changes.

**Comment by MonkeyCanCode** [comment]
> 1. Replied in [Persistence plugins in default build artifacts #125](https://github.com/polaris-catalog/polaris/discussions/125)
> 2. I think the name `metrics` is fine now. We can always change it in the future if there could be any changes.

Thanks for the feedback. I will proceed with those for now. As I am trying to get it prod ready, I will submit PR for helm chart shortly with same consistency then request you and @snazy to review (follow the similar layout as what Nessie is using for helm...but Nessie is moving toward for operator I think)

**Comment by sfc-gh-ygu** [comment]
Hi @MonkeyCanCode, can you rebase the change so that we can merge it?

**Comment by MonkeyCanCode** [comment]
> Hi @MonkeyCanCode, can you rebase the change so that we can merge it?

Done

**Review by flyrain — APPROVED** [review]

**Comment by flyrain** [comment]
Thanks @MonkeyCanCode for the PR.

**Pull request merged**